### PR TITLE
Server-side: Calculations needed to compute the client-side Differential Student-test

### PR DESCRIPTION
**Pull request by naudinlo**
## Description

Related to https://github.com/arkhn/dsPrivacy/issues/6

## Technical details

`calculationsForStudentTestDP` computes necessary calculations for one group to compute student-test on the client side with differential privacy.

## Definition of Done

- [ ] I followed the [Arkhn Code Book](https://www.notion.so/arkhn/Arkhn-Code-Book-23ac18a8af7343d0a3f61f1c9ef7400c) (I swear!).
- [ ] I have written tests for the code I added or updated.
- [ ] I have updated the documentation according to my changes.
- [ ] I have updated the deployment configuration if needed.

### CODE REVIEW

File: R/studentTestDP.R [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+#'
+#' @return a named list with 3 differentially private numeric values: the sum (Sum), the sum of squares (SumSquares) and the length of valid elements (Nvalid) of the input vector
+#' @export
+
+
+calculationsForStudentTestDP <- function(x, epsilon, x_min, x_max){
+  split_epsilon <- epsilon / 3 
+  length <- py_module$pydp_wrapper$count_PyDP$pyDP_count(x[!is.na(x)], split_epsilon)
+  sum <- py_module$pydp_wrapper$sum_PyDP$pyDP_bounded_sum(x, split_epsilon, x_min, x_max)
+  sum_squares <- py_module$pydp_wrapper$sum_PyDP$pyDP_bounded_sum(x^2, split_epsilon, x_min, x_max)
```
Comment (LaRiffle) [inline R/studentTestDP.R:17]: Here you should put the bounds of `x^2` not `x`

File: inst/DATASHIELD [hunk 2]
```diff
@@ -7,5 +7,6 @@ AggregateMethods:
   countDP,
   maxDP,
   medianDP,
-  minDP
```
Comment (LaRiffle) [inline inst/DATASHIELD:10]: I _think_ you should also declare the new function in `NAMESPACE`?

**Review by LaRiffle — CHANGES_REQUESTED** [review]
Looks good to me except for the sum of squares!

File: R/studentTestDP.R [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+#'
+#' @return a named list with 3 differentially private numeric values: the sum (Sum), the sum of squares (SumSquares) and the length of valid elements (Nvalid) of the input vector
+#' @export
+
+
+calculationsForStudentTestDP <- function(x, epsilon, x_min, x_max){
+  split_epsilon <- epsilon / 3 
+  length <- py_module$pydp_wrapper$count_PyDP$pyDP_count(x[!is.na(x)], split_epsilon)
+  sum <- py_module$pydp_wrapper$sum_PyDP$pyDP_bounded_sum(x, split_epsilon, x_min, x_max)
+  sum_squares <- py_module$pydp_wrapper$sum_PyDP$pyDP_bounded_sum(x^2, split_epsilon, x_min^2, x_max^2)
```
Comment (LaRiffle) [inline R/studentTestDP.R:17]: It is not necessary the case that `x_min^2 < x_max^2` (depending on the signs).
Suggestion : 
```
xx_min = min(x_min * x_min, x_max * x_max)
xx_max = max(x_min * x_min, x_max * x_max)
```

**Review by LaRiffle — CHANGES_REQUESTED** [review]

File: R/studentTestDP.R (hunk 3 above)
Comment (naudinlo) [reply R/studentTestDP.R]: 🤦‍♀️ stupid mistake

**Review by LaRiffle — APPROVED** [review]
🙌

**Pull request merged**